### PR TITLE
Add skip_clipping parameter to skip fastp or adapterremoval

### PR DESCRIPTION
**Pull request by prototaxites**
Adds a parameter to skip read QC using either fastp or adapterremoval, to allow the user to jump in straight to assembly if they have pre-processed reads. Also adds a check so that if clipping is skipped, `keep_phix` is true, and no host fasta or genome is specified, FASTQC_TRIMMED does not run - have also fixed the multiqc call so it can run in this case. 

I've added a new test profile (test_no_clipping.config), and have tested all combinations of skip_clipping, keep_phix and host_fasta. The other tests (test.config and test_host_rm.config) still operate as they should. 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
@nf-core-bot fix linting

File: workflows/mag.nf [hunk 1]
```diff
@@ -288,10 +291,12 @@ workflow MAG {
...
-    FASTQC_TRIMMED (
-        ch_short_reads
-    )
-    ch_versions = ch_versions.mix(FASTQC_TRIMMED.out.versions)
+    if (!(params.keep_phix && params.skip_clipping && !(params.host_genome || params.host_fasta))) {
+        FASTQC_TRIMMED (
+            ch_short_reads
+        )
+        ch_versions = ch_versions.mix(FASTQC_TRIMMED.out.versions)
+    }
```
Comment (jfy133) [inline workflows/mag.nf:299]: I don't fully understand the logic of this, why you would still want run FASTQC after removing phix? Just to get the stats of number reads left or something?

**Review by jfy133 — APPROVED** [review]
Overall looks OK to me (one question about FASTQC_TRIMMEd but that's it)

File: workflows/mag.nf (hunk 1 above)
Comment (jfy133) [reply workflows/mag.nf:299]: Fair point, I think fine is either way. I'll let @skrakau  and/or @d4straub to decide for a majority vote ;)

Comment (d4straub) [reply workflows/mag.nf:299]: Fine to me either way.

Comment (jfy133) [reply workflows/mag.nf:299]: Keep it as it is then!

**Comment by prototaxites** [comment]
@nf-core-bot fix linting

**Comment by prototaxites** [comment]
@d4straub This one is now also failing because of linting (I wasn't sure what the etiquette around merging was!) - would it be possible to merge it also?

**Pull request merged**